subgraphs/uniswap-v2/src/Factory.ts

```diff
@@ -0,0 +1,150 @@
+// import { log } from '@graphprotocol/graph-ts'
+import { log } from '@graphprotocol/graph-ts'
+import { PairCreated } from './../generated/Factory/Factory'
+import { DexAmmProtocol, LiquidityPool, Token, _HelperStore, _TokenTracker } from './../generated/schema'
+import { Pair as PairTemplate } from '../generated/templates'
+import { fetchTokenSymbol, fetchTokenName, fetchTokenDecimals } from './common/tokens'
+import { WHITELIST } from './common/Price'
+import {
+  FACTORY_ADDRESS,
+  BIGDECIMAL_ZERO,
+  ProtocolType,
+  Network,
+  BIGDECIMAL_ONE,
+  INT_ZERO
+} from './common/constants'
+
+export function handleNewPair(event: PairCreated): void {
+
+  log.debug('New Pair Created -', [])
+
+  let protocol = DexAmmProtocol.load(FACTORY_ADDRESS)
+
+  if (protocol === null) {
+    protocol = new DexAmmProtocol(FACTORY_ADDRESS)
+    protocol.name = "Uniswap v2"
+    protocol.slug = "uniswap-v2"
+    protocol.totalValueLockedUSD = BIGDECIMAL_ZERO
+    protocol.network = Network.ETHEREUM
+    protocol.type = ProtocolType.EXCHANGE
+
+    // create new ether price storage object
+    let ether = new _HelperStore('ETH')
```

Comment:
You should use named constants wherever you can as it makes it easier to maintain the code later on. Also helps with readability when there are multiple places where you'd use 'ETH' but not for the same reason.

Reply:
Okay - I made it a constant